### ISSUE TITLE
PSR12 on PHP 7.4 (typed properties + return types)

### ISSUE THREAD

**Issue by madslundholmdk** [squizlabs/PHP_CodeSniffer#2598]
```
PHP_CodeSniffer version 3.4.2 (stable) by Squiz
```
```
PHP 7.4.0beta2 (cli) (built: Aug 24 2019 10:18:17) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0-dev, Copyright (c) Zend Technologies
```

I had a file using PHP 7.4 features (typed properties + return types), like this:
```php
<?php

class Test
{
    private ?object $property = null;

    public function test(): ?object
    {
        return $this->property;
    }
}
```

The nullable typed property and the return type was parsed as a shorthand if else statement.
And therefore it output the following error:
```
FILE: Test.php
----------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------------------------------------------------
 5  | ERROR | [x] Expected at least 1 space after "?"; 0 found
    |       |     (PSR12.Operators.OperatorSpacing.NoSpaceAfter)
 7  | ERROR | [x] Expected at least 1 space before ":"; 0 found
    |       |     (PSR12.Operators.OperatorSpacing.NoSpaceBefore)
----------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------------------------
```
which are incorrect due to [PSR12](https://www.php-fig.org/psr/psr-12/#45-method-and-function-arguments)

**Comment by jrfnl** [comment]
AFAICS this was addressed via #2413 which will be included in the upcoming PHPCS 3.5.0 release.

**Comment by gsherwood** [comment]
Yes, support for typed properties has been added to master and will be released in version 3.5.0. I can confirm that the PSR12 standard in master does not produce the errors you are seeing.
